data/api/server-server/definitions/event-schemas/m.direct_to_device.yaml

```
@@ -43,7 +43,7 @@ allOf:
           message_id:
             type: string
             description: Unique ID for the message, used for idempotence.  Arbitrary
-              utf8 string, of maximum length 32 codepoints.
+              utf8 string, of maximum length 32 bytes.
```

this is a pre-existing problem, but `message_id` should *not* be a UTF-8 string (or rather, it does not need to be one and should not be interpreted as one). It should be a regular unicode string, just like all the other strings in matrix. Assuming it is then sent over an HTTP request which uses a JSON body with UTF-8 encoding, it will be encoded as UTF-8.

In particular: if it were a UTF-8 string, then bytes such as `0xC0` would be invalid. In practice, `0xC0` is a valid value in this string; it is ["LATIN CAPITAL LETTER A WITH GRAVE"](https://www.fileformat.info/info/unicode/char/00c0/index.htm).

In short, this text needs rewording, rather than simply s/codepoints/bytes/:

```suggestion
            description: Unique ID for the message, used for idempotence.  Arbitrary
              string, whose UTF-8 encoding must not exceed 32 bytes.
```